cmd/stop_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright © 2019 IBM Corporation and others.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd_test
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"os"
+	"testing"
+
+	"github.com/appsody/appsody/cmd/cmdtest"
+)
+
+func TestStopWithoutName(t *testing.T) {
+	// first add the test repo index
+	_, cleanup, err := cmdtest.AddLocalFileRepo("LocalTestRepo", "testdata/index.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+	// create a temporary dir to create the project and run the test
+	projectDir, err := ioutil.TempDir("", "appsody-stop-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(projectDir)
+	log.Println("Created project dir: " + projectDir)
+
+	// appsody init nodejs-express
+	_, err = cmdtest.RunAppsodyCmdExec([]string{"init", "nodejs-express"}, projectDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// appsody run
+	runChannel := make(chan error)
+	go func() {
+		_, err = cmdtest.RunAppsodyCmdExec([]string{"run"}, projectDir)
+		runChannel <- err
+	}()
+
+	// defer the appsody stop to close the docker container
+	defer func() {
+
+		stopOutput, errStop := cmdtest.RunAppsodyCmdExec([]string{"stop"}, projectDir)
+
+		//docker[stop appsody-stop-test
+		if !strings.Contains(stopOutput, "docker[stop appsody-stop-test") {
```

Comment:
Checking that the stop command ran is okay, but I think we really should run a `docker ps` to verify the container was stopped.